src/Components/OATGraphViewer/OATGraphViewer.tsx

```diff
@@ -90,6 +93,45 @@ const OATGraphViewer = ({ state, dispatch }: OATGraphProps) => {
     const [showComponents, setShowComponents] = useState(true);
     const [rfInstance, setRfInstance] = useState(null);
 
+    const dagreGraph = new dagre.graphlib.Graph();
+    dagreGraph.setDefaultEdgeLabel(() => ({}));
+
+    const getLayoutedElements = (elements, direction = 'TB') => {
+        const isHorizontal = direction === 'LR';
+        dagreGraph.setGraph({ rankdir: direction });
+
+        elements.forEach((el) => {
+            if (isNode(el)) {
+                dagreGraph.setNode(el.id, {
+                    width: nodeWidth,
+                    height: nodeHeight
+                });
+            } else {
+                dagreGraph.setEdge(el.source, el.target);
+            }
+        });
+
+        dagre.layout(dagreGraph);
+
+        return elements.map((el) => {
+            if (isNode(el)) {
+                const nodeWithPosition = dagreGraph.node(el.id);
+                el.targetPosition = isHorizontal ? 'left' : 'top';
+                el.sourcePosition = isHorizontal ? 'right' : 'bottom';
+
+                el.position = {
+                    x:
+                        nodeWithPosition.x -
+                        nodeWidth / 2 +
+                        Math.random() / 1000,
```

Comment:
Can you help me understand why random is required here? Non-deterministic code always gets my interest 😄

Reply:
![image](https://user-images.githubusercontent.com/97867829/171291383-bb52594e-2c67-44f8-a76d-307b8f2073ba.png)
Its a little hack I found necessary for Flow renderer V9, initially was using the examples from V10 and it didn't worked, it looks they improve it and its no longer required for any V10 implementation, not our case though